native/java/org/jpype/pkg/JPypePackageManager.java

```diff
@@ -383,6 +383,26 @@ private static void getJarContents(Map<String, URI> out, String packageName)
       while (resources.hasMoreElements())
       {
         URI resource = resources.nextElement().toURI();
+
+        // Handle MRJAR format
+        //   MRJAR may not report every directory but instead just the overlay.
+        //   So we need to find the original and interogate it first before
+        //   checking the version specific part.  Worst case we collect the
+        //   contents twice.
+        String schemePart = resource.getSchemeSpecificPart();
+        int index = schemePart.indexOf("!");
+        if (index != -1)
+        {
+          if (schemePart.substring(index + 1).startsWith("/META-INF/versions/"))
+          {
+            int index2 = schemePart.indexOf('/', index + 20);
+            schemePart = schemePart.substring(0, index + 1) + schemePart.substring(index2);
+            URI resource2 = new URI(resource.getScheme() + ":" + schemePart);
+            Path path3 = getPath(resource2);
+            collectContents(out, path3);
```

Comment:
why can't we break here, after treating the special case? If there are more jars to inspect?

Reply:
We have to search both the non version case and the versioned case.  It is a weird format.   Mind you we could use this format to make org.jpype run on many versions but the structure of setting one of these up is not that fun.  So instead i use reflection when we have version issues.